timemachine/cpp/src/integrator.hpp

```diff
@@ -7,53 +7,31 @@ namespace timemachine {
 class Integrator {
 
 public:
+    virtual ~Integrator(){};
 
-    virtual ~Integrator() {};
-
-    virtual void step_fwd(
-        double *d_x_t,
-        double *d_v_t,
-        unsigned long long *d_du_dx_t,
-        double *d_box_t_,
-        cudaStream_t stream
-    ) = 0;
-
+    virtual void
+    step_fwd(double *d_x_t, double *d_v_t, unsigned long long *d_du_dx_t, double *d_box_t_, cudaStream_t stream) = 0;
 };
 
 class LangevinIntegrator : public Integrator {
 
 private:
-
     double dt_;
     double N_;
     double ca_;
     double *d_cbs_;
     double *d_ccs_;
     double *d_noise_;
 
-    curandGenerator_t  cr_rng_;
+    curandGenerator_t cr_rng_;
 
 public:
-
-    LangevinIntegrator(
-        int N,
-        double dt,
-        double ca,
-        const double *h_cbs,
-        const double *h_ccs,
-        int seed
-    );
+    LangevinIntegrator(int N, double dt, double ca, const double *h_cbs, const double *h_ccs, int seed);
 
     virtual ~LangevinIntegrator();
 
     virtual void step_fwd(
-        double *d_x_t,
-        double *d_v_t,
-        unsigned long long *d_du_dx_t,
-        double *d_box_t_,
-        cudaStream_t stream
-    ) override;
-
+        double *d_x_t, double *d_v_t, unsigned long long *d_du_dx_t, double *d_box_t_, cudaStream_t stream) override;
```

Comment:
I also prefer Black's behavior here; I just did a little more looking and found [ExperimentalAutoDetectBinPacking](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#:~:text=experimentalautodetectbinpacking), which seems like it might be exactly what we want. But it comes with a disclaimer:

> NOTE: This is an experimental flag, that might go away or be renamed. Do not use this in config files, etc. Use at your own risk.

It might still be worth trying out?

Reply:
Hmm

>Each call can be bin-packed, one-per-line or inconclusive. If it is inconclusive, e.g. completely on one line, but a decision needs to be made, clang-format analyzes whether there are other bin-packed cases in the input file and act accordingly.

doesn't sound ideal to me -- would prefer to have the formatter apply an unambiguous rule (and ideally allow for a manual toggle), but would not want the formatter to make complicated decisions based on the rest of the file.

I'd be in favor of keeping this as you currently have it!